### PR TITLE
chore: update transaction for api edit skills

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] Update db transaction for update skills api

### CODE REVIEW

File: pkg/handler/employee/employee.go [hunk 1]
```diff
@@ -442,27 +444,80 @@ func (h *handler) UpdateSkills(c *gin.Context) {
...
 	if err != nil {
-		if errors.Is(err, gorm.ErrRecordNotFound) {
-			l.Error(ErrEmployeeNotFound, "error employee not found")
-			c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, ErrEmployeeNotFound, body))
+		l.Error(err, "error when finding employee")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, body))
+		return
+	}
+
+	// 3. update in transaction
```
Comment (namnhce) [inline pkg/handler/employee/employee.go:462]:
```suggestion
	// 3. begin transaction
```

File: pkg/handler/employee/employee.go [hunk 2]
```diff
@@ -442,27 +444,80 @@ func (h *handler) UpdateSkills(c *gin.Context) {
...
-	// 3. update info and return
-	rs, err := h.store.Employee.UpdateSkills(h.repo.DB(), employee.UpdateSkillsInput{
-		Positions: body.Positions,
-		Chapter:   body.Chapter,
-		Seniority: body.Seniority,
-		Stacks:    body.Stacks,
-	}, employeeID)
+	// check employee existence
+	exist, err := h.store.Employee.Exist(h.repo.DB(), employeeID)
+	if !exist {
```
Comment (namnhce) [inline pkg/handler/employee/employee.go]: check error first

File: pkg/store/employeeposition/interface.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	"gorm.io/gorm"
+
+	"github.com/dwarvesf/fortress-api/pkg/model"
+)
+
+type IStore interface {
+	All(db *gorm.DB) (employeePosition []*model.EmployeePosition, err error)
+	One(db *gorm.DB, id model.UUID) (employeePosition *model.EmployeePosition, err error)
+	Create(db *gorm.DB, employeePosition *model.EmployeePosition) (*model.EmployeePosition, error)
+	SorfDelete(db *gorm.DB, employeeID string) (err error)
```
Comment (namnhce) [inline pkg/store/employeeposition/interface.go:13]: Delete

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/handler/employee/employee.go (hunk 1 above)
Comment (datphamcode295) [reply pkg/handler/employee/employee.go]: updated

File: pkg/handler/employee/employee.go (hunk 2 above)
Comment (datphamcode295) [reply pkg/handler/employee/employee.go]: updated

File: pkg/store/employeeposition/interface.go (hunk 3 above)
Comment (datphamcode295) [reply pkg/store/employeeposition/interface.go]: updated to HardDelete

**Review by namnhce — APPROVED** [review]

**Pull request merged**